src/codegen/traverse.lisp

```diff
@@ -7,11 +7,19 @@
    (#:util #:coalton-impl/util))
   (:export
    #:action
+   #:count-applications
+   #:count-nodes
+   #:*traverse*
    #:traverse
    #:traverse-with-binding-list))
 
 (in-package #:coalton-impl/codegen/traverse)
 
+(defvar *traverse* nil
+  "The recursive function reference for the current traversal. This will
+be bound for the extent of any traversal. Inside of an
+`action`, `(funcall *traverse* node)` may be used to invoke recursion.")
```

Comment:
Most changes in this file are renaming and documentation updates to reflect this

Reply:
If this has no meaning outside of the traversal, I recommend making it unbound.

```
(defvar *traverse*)
(setf (documentation '*traverse 'variable)
      "doc string")
```

It's a little obtuse but it ensures `*traverse*` will error when evaluated outside of its context.

Reply:
Done: https://github.com/coalton-lang/coalton/pull/1193/files?diff=unified&w=0#diff-6613c23f2ebfe4977201bab1561ab3e597cdc43410389416ddfaae8a03194f98R19-R24